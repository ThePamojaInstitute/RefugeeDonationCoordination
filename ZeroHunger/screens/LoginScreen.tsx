import { StyleSheet, Text, View, KeyboardAvoidingView, TextInput, TouchableOpacity } from 'react-native'
import React, {useState} from 'react'

const LoginScreen = () => {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');


  return (
    <KeyboardAvoidingView behavior='height' style = {styles.container}>
    
        <View style={styles.inputContainer}>
            <TextInput style = {styles.input}
                placeholder='Email'
                value = {email}
                onChangeText={text => setEmail(text)}
            />            
            <TextInput style = {styles.input}
                placeholder='Password'
                secureTextEntry
                value = {password}
                onChangeText={text => setPassword(text)}
            />
        </View>

        <View style = {styles.buttonContainer}>
            <TouchableOpacity style = {styles.button} onPress={() => {}} >
                <Text style={styles.buttonText}> Login </Text>

            </TouchableOpacity>

            <TouchableOpacity style = {[styles.button, styles.buttonOutline]} onPress={() => {}} >
                <Text style={styles.buttonOutlineText}> Register </Text>

            </TouchableOpacity>
        </View>

    </KeyboardAvoidingView>
  )
}

export default LoginScreen

const styles = StyleSheet.create
({
    container: 
    {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    },
    inputContainer:
    {
        width: '80%'
    },
    input:
    {
        backgroundColor: 'white',
        paddingHorizontal: 15,
        paddingVertical: 10,
        borderRadius: 10,
        marginTop: 5
    },
    buttonContainer:
    {
        width: '60%',
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 20

    },
    button:
    {
        backgroundColor: '#0782f9',
        width: '100%',
        padding: 15,
        borderRadius: 10,
        alignItems: 'center'
    },
    buttonOutline:
    {
        backgroundColor: 'white',
        marginTop: 5,
        borderColor: '#0782f9',
        borderWidth: 2
    },
    buttonOutlineText:
    {
        color: '#0782f9',
        fontWeight: '700',
        fontSize: 16
    },
    buttonText:
    {
        color: 'white',
        fontWeight: '700',
        fontSize: 16
    }

})