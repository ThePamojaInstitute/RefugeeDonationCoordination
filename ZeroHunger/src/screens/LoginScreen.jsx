import React from "react";
import { StyleSheet, Text, View, TextInput, Button, TouchableOpacity } from "react-native";

// export default function App() {
const Loginscreen = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.login}>Zero Hunger</Text>
      <View style={styles.inputView}>
        <TextInput
          style={styles.input}
          placeholder="Username"
          placeholderTextColor="#000000"
        /> 
      </View> 
      <View style={styles.inputView}>
        <TextInput
          style={styles.input}
          placeholder="Password"
          placeholderTextColor="#000000"
          secureTextEntry={true}
        /> 
      </View> 
      <TouchableOpacity /*onPress={}*/>
        <Text style={styles.forgotBtn}>Forgot password?</Text> 
      </TouchableOpacity> 
      <TouchableOpacity style={styles.loginBtn} /*onPress={}*/>
        <Text style={styles.loginBtnText}>Login</Text> 
      </TouchableOpacity> 
      <TouchableOpacity style={styles.signUpBtn} /*onPress={}*/>
        <Text style={styles.signUpBtnText}>Sign Up</Text>
      </TouchableOpacity>
    </View> 
  );
}

export default LoginScreen

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  inputView: {
    backgroundColor: "#D3D3D3",
    borderRadius: 30,
    width: "70%",
    height: 45,
    marginBottom: 10,
    marginTop: 5,
  },
  login: {
    color: "#000000",
    height: 140,
    fontSize: 48
  },
  input: {
    flex: 1,
    padding: 10,
    marginLeft: 10,
    fontSize: 15,
  },
  forgotBtn: {
    height: 16,
    marginBottom: 30,
    marginLeft: 150,
    fontSize: 12,
  },
  loginBtn: {
    title: "Login",
    width: "85%",
    borderRadius: 25,
    marginTop: 30,
    height: 50,
    alignItems: "center",
    backgroundColor: "#6A6A6A",
  },
  loginBtnText: {
    color: "#FFFFFF",
    padding: 15,
    marginLeft: 10,
    fontSize: 15,
  },
  signUpBtn: {
    title: "Sign Up",
    width: "85%",
    borderRadius: 25,
    marginTop: 15,
    height: 50,
    alignItems: "center",
    backgroundColor: "#A9A9A9",
  },
  signUpBtnText: {
      color: "#FFFFFF",
      padding: 15,
      marginLeft: 10,
      fontSize: 15,
  }
});