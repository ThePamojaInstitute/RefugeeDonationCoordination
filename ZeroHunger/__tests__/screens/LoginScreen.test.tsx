import LoginScreen from "../../src/screens/LoginScreen";
import { render, screen } from '@testing-library/react-native';
import React from "react";


test('matches snapshot', () => {
    const rendered = render(<LoginScreen navigation={null} />).toJSON();

    expect(rendered).toMatchSnapshot();
});