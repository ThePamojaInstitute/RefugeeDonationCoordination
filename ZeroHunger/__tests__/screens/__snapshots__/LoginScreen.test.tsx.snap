// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="testId"
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 48,
        "height": 140,
      }
    }
  >
    Zero Hunger
  </Text>
  <Text />
  <View
    style={
      {
        "backgroundColor": "#D3D3D3",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 10,
        "marginTop": 5,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChange={[Function]}
      placeholder="Username"
      placeholderTextColor="#000000"
      style={
        {
          "flex": 1,
          "fontSize": 15,
          "marginLeft": 10,
          "padding": 10,
        }
      }
    />
  </View>
  <View
    style={
      {
        "backgroundColor": "#D3D3D3",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 10,
        "marginTop": 5,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChange={[Function]}
      placeholder="Password"
      placeholderTextColor="#000000"
      secureTextEntry={true}
      style={
        {
          "flex": 1,
          "fontSize": 15,
          "marginLeft": 10,
          "padding": 10,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 12,
          "height": 16,
          "marginBottom": 30,
          "marginLeft": 150,
        }
      }
    >
      Forgot password?
    </Text>
  </View>
  <Text
    style={
      {
        "color": "red",
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#6A6A6A",
        "borderRadius": 25,
        "height": 50,
        "marginTop": 30,
        "opacity": 1,
        "title": "Login",
        "width": "85%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 15,
          "marginLeft": 10,
          "padding": 15,
        }
      }
    >
      Login
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#A9A9A9",
        "borderRadius": 25,
        "height": 50,
        "marginTop": 15,
        "opacity": 1,
        "title": "Sign Up",
        "width": "85%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 15,
          "marginLeft": 10,
          "padding": 15,
        }
      }
    >
      Sign Up
    </Text>
  </View>
</View>
`;
